freud/environment.pyx

```diff
@@ -512,7 +512,18 @@ cdef class _MatchEnv(_PairCompute):
 
 cdef class EnvironmentCluster(_MatchEnv):
     r"""Clusters particles according to whether their local environments match
-    or not, according to various shape matching metrics.
+    or not, using various shape matching metrics defined in :cite:`Teich2019`.
+
+    Note that two sets of neighbor lists are used. :code:`'env_neighbors'`
+    defines the query particles' environments. :code:`'neighbors'`
+    defines the neighborhood where query particles' environments are compared
+    to find the common clusters.
+
+    For example, with :code:`'env_neighbors = {num_neighbors: 12}'` and
+    :code:`'neighbors = {'num_neighbors': 15}'`, the :code:`compute` function
+    would construct an environment that contains 12 neighbors for a query
+    particle, and compare it with the environments of the particle's 15 nearest
+    neighbors.
```

Comment:
I think using "graph" makes it more complex, because we don't mention graph anywhere else. Do you want to use graph language because of the algorithm?

Reply:
Yes, basically the algorithm is graph traversal while removing edges between points with different environments and then doing clustering on the resulting graph. This suggestion can be disregarded though if that's too confusing